components/SingleClassView/Module/Module.tsx

```diff
@@ -0,0 +1,400 @@
+            </div>
+            <div className={styles.buttonContainer}>
+              <button onClick={handleCancel} className={styles.cancel}>
+                Cancel
+              </button>
+              <button onClick={handleYes} className={styles.submit}>
+                Yes
+              </button>
+            </div>
+          </div>
+        </div>
+      ) : null}
+    </div>
+  );
+};
+
+type AccordionModuleProps = {
+  module: Module;
+  numModules: number;
+};
+
+const AccordionModule: React.FC<AccordionModuleProps> = ({ module, numModules }) => {
+  const api = useContext(APIContext);
+  const [lessons, setLessons] = useState<Item[]>([]);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const [update, setUpdate] = useState(false);
+  const [name, setName] = useState(module.name);
+  const [position, setPosition] = useState(module.position);
+  const [add, setAdd] = useState(false);
+  const [itemName, setItemName] = useState("");
+  const [deleteItem, setDeleteItem] = useState(false);
+  const [itemLink, setItemLink] = useState("");
+
+  useEffect(() => {
+    (async () => {
+      const res = await api.getModuleItems(module.moduleId);
+      setLessons(res);
+    })();
+  }, []);
+
+  if (!lessons) return <CustomLoader />;
+
+  // eslint-disable-next-line
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  // eslint-disable-next-line
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleRename: VoidFunction = () => {
+    setUpdate(!update);
+    handleClose();
+  };
+
+  const handleAdd: VoidFunction = () => {
+    setAdd(!add);
+    handleClose();
+  };
+
+  const handleDelete: VoidFunction = async () => {
+    setDeleteItem(true);
+    handleClose();
+  };
+
+  const handleSubmit = async (): Promise<void> => {
+    const inModule = {
+      moduleId: module.moduleId,
+      classId: module.classId,
+      name: name,
+      position: position,
+    };
+    await api.updateModule(module.moduleId, inModule);
+    setUpdate(!update);
+  };
+
+  const handleCancel = async (): Promise<void> => {
+    setName(module.name);
+    setUpdate(false);
+    setAdd(false);
+  };
+
+  const handleDeleteSubmit: VoidFunction = async () => {
+    await api.deleteModule(module.moduleId);
+    await setDeleteItem(false);
+  };
+
+  const handleAddSubmit: VoidFunction = async () => {
+    const lesson = {
+      title: itemName,
+      link: itemLink,
+      moduleId: module.moduleId,
+      itemId: "",
+    };
+    await api.createItem(module.moduleId, lesson);
+    setAdd(!add);
+    setLessons([...lessons, lesson]);
+  };
+
+  const handleMoveUp: VoidFunction = () => {
+    if (position > 0) {
+      setPosition(position - 1);
+    }
+    handleClose();
+  };
+
+  const handleMoveDown: VoidFunction = () => {
+    if (position < numModules) {
+      setPosition(position + 1);
```

Comment:
you have to also reconfigure everything else above and below for this to work correctly

Reply:
Same for move above

Reply:
You will need to store this in an array so you can handle the move up and down in an ordered fashion

Reply:
Also, you need to make a PATCH call with the modules to save the updated data in the backend

Reply:
addressed in slack